src/buildblock/ProjDataFromStream.cxx

```diff
@@ -371,6 +461,93 @@ ProjDataFromStream::set_viewgram(const Viewgram<float>& v)
 }
 
 
+std::vector<std::streamoff>
+ProjDataFromStream::get_offsets_bin(const int segment_num,
```

Comment:
I think it might be best to rewrite this in terms of get_offsets. That way the ugly code is in only one place? Or we could rewrite get_offsets in terms of this one of course.